x-pack/plugins/apm/common/search_strategies/latency_correlations/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  HistogramItem,
+  RawResponseBase,
+  SearchStrategyClientParams,
+} from '../types';
+
+export interface LatencyCorrelation {
+  correlation: number;
+  fieldName: string;
+  fieldValue: string;
+  histogram: HistogramItem[];
+  ksTest: number;
+}
+
+// Basic type guard for array of LatencyCorrelation
+export function isLatencyCorrelations(
+  arg: unknown
```

Comment:
The naming seems (`arg`) is very generic. What is it? And why is the type `unknown`? Don't we have any control over the input here?

Reply:
These are meant to be type guards, they combine runtime duck typing with TS. It's quite generic on purpose to narrow down any given input and find out if it matches the type. A lot of examples use `any` but we'd want to avoid that so it's `unknown`. https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
So this is coming from the user and not our own application and we are worried about tampering? 
If so, we already use io-ts for this to validate input at the edges.

Reply:
Summarizing the Slack discussion we had: Since this is code we control coming from the server, the type guard is a bit overkill and I changed it back to just inline checks in the consuming components in c52158b.